pingcap/pd
# api: add initialized flag in cluster status

Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve? 
When initializing the cluster, PD needs to spend some time to add replica to the first region. If tidb-server is started before finish adding replica, the data may be written to simple replica.

### What is changed and how it works?
PD adds an flag of whether the region is initialized or not. Later `tidb-ansible` need to check whether it needs to wait for a while before starting tidb.

### Check List 
Tests 
 - Unit test
 - Manual test (add detailed scripts or steps below)
start a cluster then check the output of `curl http://127.0.0.1:2379/pd/api/v1/cluster/status`

Code changes
 - Has HTTP API interfaces change

Side effects
 - Breaking backward compatibility: Before this, we allow to config `max-replicas=3` and only start 1 tikv-server. It will be required to set `max-replicas=1` to start the cluster.

Related changes
 - Need to cherry-pick to the release branch: cherry-pick to release-3.0
 - Need to update the `tidb-ansible` repository: `ansible` need to check the status before starting tidb-server
 - Need to be included in the release notes
